enhancements/api-review/custom-tags-aws.md

```diff
@@ -48,77 +53,178 @@ Motivations include but are not limited to:
+                type: string
+                maxLength: 256
+                minLength: 1
+                pattern: ^[0-9A-Za-z_.:/=+-@]+$
+```
+
 ### User Stories
 
 https://issues.redhat.com/browse/SDE-1146 - IAM users and roles can only operate on resources with specific tags
 As a security-conscious ROSA customer, I want to restrict the permissions granted to Red Hat in my AWS account by using
 AWS resource tags.
 
+### API Extensions
+
+This proposal edits `Infrastructure.config.openshift.io/v1` type
+
+### Operational Aspects of API Extensions
+
+NA
+
 ### Risks and Mitigations
 
+#### Failure Modes
+
+NA
+
+#### Support Procedures
+
+NA
+
 ## Design Details
 
+User can add tags during installation for creation of resources with tags. Installer creates infrastructure manifests with `resourceTags` to `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io`.
+Cluster config operator reconciles Infrastructure object, validates and updates `resourcesTags` from `.spec.platformSpec.aws` to  `.status.platformStatus.aws`.
+Operators that operate on aws resources must merge the tags from `.spec.platformSpec.aws`, `.status.platformStatus.aws` and individual component tags.
+
+The values from `status.platformStatus.aws` will used to only support downgrade.
+
+In case of precedence conflict or errors, the same will be reported using `Events` by the operators.
+
+`.status.platformStatus.aws` can be deprecated in the future versions when there is a CRD validation possible to avoid removal of userTags from `.spec.platformSpec.aws`.
+
 ### Test Plan
 
-TBD
+Update the `resourceTags` testcases to check successful update from Infrastructure resource.
 
 ### Graduation Criteria
 
-TBD
-
 #### Dev Preview -> Tech Preview
 
-TBD
+NA
 
 #### Tech Preview -> GA
 
-TBD
+- Upgrade/downgrade testing
+- Sufficient time for feedback
+- Available by default
+- Stress testing for scaling and tag update scenarios
 
 #### Removing a deprecated feature
 
-This enhancement does not remove or deprecate any features.
+`experimentalPropagateUserTags` field will be set for deprecation.
+`.status.platformStatus.aws` will be set for deprecation.
 
 ### Upgrade / Downgrade Strategy
 
 On upgrade:
 
-The new status field won't be populated since it is only populated by the installer and that can't have happened if the cluster was installed from a prior version. Components that consume the new field should take no action since they will see no additional tags.
+On upgrade from version supporting `experimentalPropagateUserTags`, openshift components that consume the new field should merge `.status.platformStatus.aws` and `.spec.platformSpec.aws`.
+`.spec.platformSpec.aws` can be updated or changed which will override the values from status and a new `.status.platformStatus.aws` will be set.
+
+For installer configuration compatibility, configuration with `experimentalPropagateUserTags` should be supported.
```

Comment:
How is this going to work? If the default behavior is to propagate user tags and the default of `experimentalPropagateUserTags` is false, that is an unsupported configuration.

Reply:
This behaviour is updated. The idea was to have versioning of tags to identify the creator tool.